src/services/signatureHelp.ts

```diff
@@ -307,10 +307,13 @@ namespace ts.SignatureHelp {
         // for optional function condition.
         const nonNullableContextualType = contextualType.getNonNullableType();
 
-        const signatures = nonNullableContextualType.getCallSignatures();
-        if (signatures.length !== 1) return undefined;
+        const symbol = nonNullableContextualType.symbol;
+        if (symbol === undefined) return undefined;
 
-        const invocation: ContextualInvocation = { kind: InvocationKind.Contextual, signature: first(signatures), node: startingToken, symbol: chooseBetterSymbol(nonNullableContextualType.symbol) };
+        const signature = firstOrUndefined(nonNullableContextualType.getCallSignatures());

```

Comment:
If I'm not mistaken, it looks like this is also changing some other behavior for examples like the following:

```ts
interface MultiSignatureCallback {
    (x: string): string;
    (x: number): number;
    (x: string | number): string | number;
}

declare function doIt(callback: MultiSignatureCallback): void;

doIt((/*$*/) => {});
```

We should add some sort of test for this. Technically that function will never be contextually typed, but it is often helpful to see the signature help when typing it out. So I think it's useful to show *some* signature in these cases.

If we want to do that right, I think we should probably switch to the last signature rather than the first. We generally make the assumption that the last signature is always the most-general.

```suggestion
        const signature = lastOrUndefined(nonNullableContextualType.getCallSignatures());
```